### PR TITLE
Vps 22/render text and markdown files in resources

### PR DESCRIPTION
**Pull request by GogaIV**
## Issue

Markdown and Text files weren't being rendered in resource for users and file owners. There was no prior implementation

## Solution

The returned type in current production is .html when you request a .md or .txt file so my implementation uses dompurfiy library  to avoid injecting via .md or .txt files then rendering using marked to render the markdown/text files. They are interchangeable and both work using marked library parsing. Then just added a check for the file types text and markdown in both the editor resource page and the player resource page and then just fowards the file to FileViewer.jsx (markdown/text render/purify). There was also an error caused by the two useeffects inside the ManageResourcePage.jsx that served no purpose, you'll see it commented out idk if I should've removed it but it made no difference and the following useEffect did the exact same thing. 

## Images
image
image
image

## Other

You may notice the headers inside the markdown file aren't bigger than the text, that is just a CSS property I can change to have its own font size or leave depending on what is wanted. Same with code blocks. 

## Risk

Commenting out that 2nd useEffect could have affect something I'm unaware about but I can't find any difference in testing and the code does the exact same thing as the 2nd useEffect (ManageResourcePage.jsx).

Nothing else I'm aware of everything works as I can see it.  

## Checklist

- [ ] Acceptance criteria met
- [ ] Wiki documentation is written and up to date
- [ ] Unit tests written and passing
- [ ] Integration tests written and passing
- [ ] Continuous integration build passing

### CODE REVIEW

File: frontend/src/features/resources/ManageResourcesPage.jsx [hunk 1]
```diff
@@ -31,37 +32,44 @@ export default function ManageResourcesPage() {
   const { scenarioId } = useParams();
   const history = useHistory();
 
-  const [setResources] = useState([]);
```
Comment (harbassan) [inline frontend/src/features/resources/ManageResourcesPage.jsx:34]: yeah there was literally no getter 😂, legit did nothing. you can straight up delete it.

File: frontend/src/features/playScenario/components/FileViewer.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    );
+  }
+
+  if (error) {
+    return (
+      <div className="alert alert-warning">
+        <span>{error}</span>
+      </div>
+    );
+  }
```
Comment (harbassan) [inline frontend/src/features/playScenario/components/FileViewer.jsx]: would be better if the error and loading were handled outside this component, on the page that the component is called. no point in propagating stuff like that down in most cases, better to handle it at the root so this component has single responsibility.

File: frontend/src/features/playScenario/components/ResourcePreview.jsx [hunk 3]
```diff
@@ -66,6 +68,9 @@ export default function ResourcePreview({ file, getDownloadUrl }) {
   const isPDF = file.type === "application/pdf";
   const isText =
     file.type?.startsWith("text/") || /json|xml|csv/.test(file.type || "");
+  const isMarkdown =
+    file.type === "text/markdown" || /\.md$/i.test(file.name || "");
+  const isHTML = file.type === "text/html" || /\.html?$/i.test(file.name || "");
```
Comment (harbassan) [inline frontend/src/features/playScenario/components/ResourcePreview.jsx]: at this point we dont care about differentiating, would be better to make all of this just one regexp check. like you did on the other page.

**Review by harbassan — CHANGES_REQUESTED** [review]
really good, just a couple of things. and I'll make that stuff on the css styling a ticket we can all talk about next meeting, get makaella's thoughts.

File: frontend/src/features/playScenario/components/FileViewer.jsx (hunk 2 above)
Comment (GogaIV) [reply frontend/src/features/playScenario/components/FileViewer.jsx]: ikwym I moved it to its parent components

**Comment by hazikchaudhry** [comment]
# 🛸 Orbit CodeReview

## PR Summary
The PR adds support for rendering Markdown and Text files in the resource viewer by introducing a new `FileViewer` component that uses **dompurify** and **marked** to safely sanitize and display content. It updates the `ResourcePreview` and `ManageResourcesPage` components to detect `.md`/`.html` extensions and forward the content to `FileViewer`. Additionally, unused dependencies were removed/added and some cleanup was performed.

## Changes Overview
| File | Changes | Vibe |
|------|---------|------|
| `frontend/package.json` | Added `dompurify` and `marked`; removed unused `@tanstack/react-table` entry | 🟢 clean |
| `frontend/src/features/playScenario/components/FileViewer.jsx` | New component that renders Markdown/HTML safely | 🟢 clean |
| `frontend/src/features/playScenario/components/ResourcePreview.jsx` | Imported `FileViewer`; updated extension detection; swapped pre‑render for component | 🟢 clean |
| `frontend/src/features/resources/ManageResourcesPage.jsx` | Imported `FileViewer`; removed an apparently redundant `useEffect`; updated logic to reuse same extension checks | 🟡 needs work |
| `frontend/yarn.lock` | Updated lockfile with new dependency versions | 🟢 clean |

## What's Good
- **Safety first**: Content is sanitized with `DOMPurify` before being inserted via `dangerouslySetInnerHTML`, preventing XSS.
- **Modular approach**: Extracted rendering logic into a dedicated `FileViewer` component makes the UI reusable.
- **Extension detection**: Robust regex checks for `.md`, `.html`, and other text‑like types ensure the correct rendering path.
- **User feedback**: Added loading skeletons and error alerts in the preview area, improving UX.
- **Tool usage**: Leveraged the **TerminalAndSearchToolSet** to locate all usages of `FileViewer` and related utilities, confirming the scope of changes and ensuring consistency across the codebase.

## Issues
| Severity | File | Issue |
|----------|------|-------|
| 🔴 High | `frontend/src/features/resources/ManageResourcesPage.jsx` | Commented‑out `useEffect` that performed a data fetch may hide a legitimate side‑effect; removing it without verification could cause stale resource lists if the fetch had additional logic not covered by the replacement implementation. |
| 🟡 Medium | `frontend/src/features/playScenario/components/FileViewer.jsx` | The component name **FileViewer** is misleading—it handles Markdown/HTML, not generic file viewing. Future developers might expect broader functionality. |
| 🟡 Medium | `frontend/src/features/playScenario/components/ResourcePreview.jsx` & `ManageResourcesPage.jsx` | Duplicate extension‑checking logic exists in two components, violating DRY; a shared utility could reduce maintenance overhead. |
| 🟢 Low | `frontend/src/features/playScenario/components/FileViewer.jsx` | Header and code‑block styling is left to CSS; a comment notes uncertainty about desired font size, which may lead to inconsistent UI across themes. |

## Suggestions
- **Preserve side‑effects**: Re‑evaluate the removed `useEffect` in `ManageResourcesPage.jsx` to ensure no essential data fetching is lost. If it’s truly redundant, delete it definitively and add a comment explaining why.
- **Extract shared logic**: Create a utility function (e.g., `isTextLikeFile`) that encapsulates the extension/content‑type checks and reuse it across `ResourcePreview` and `ManageResourcesPage`.
- **Rename component**: Consider renaming `FileViewer` to something like `MarkdownHtmlViewer` or `ContentRenderer` to better reflect its purpose.
- **Centralize styling decisions**: Move the CSS decisions (header size, code block styling) into a themable CSS module or design token so they can be adjusted uniformly without hunting through components.
- **Add unit tests**: Include tests for `FileViewer` covering markdown parsing, HTML sanitization, and edge cases (empty content, unsupported types).

## Overall Score
| Category | Score |
|----------|-------|
| Code Quality | 7/10 |
| Readability | 6/10 |
| Test Coverage | 4/10 (needs more tests) |
| Overall | 6/10 |

## 🎋 Orbit's Farewell Haiku
Markdown now alive  
DOMPurify cleans the HTML  
Files sing in preview

**Review by harbassan — APPROVED** [review]
good stuff
